src/main/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/service/GroupsService.kt

```diff
@@ -21,38 +25,48 @@ import uk.gov.justice.digital.hmpps.externalusersapi.security.MaintainUserCheck
 @Transactional(readOnly = true)
 class GroupsService(
   private val groupRepository: GroupRepository,
-  private val maintainUserCheck: MaintainUserCheck,
   private val childGroupRepository: ChildGroupRepository,
   private val telemetryClient: TelemetryClient,
   private val authenticationFacade: AuthenticationFacade,
   private val userRepository: UserRepository,
   private val userGroupService: UserGroupService,
+  private val groupAssignableRoleRepository: GroupAssignableRoleRepository,
+  private val maintainUserCheck: MaintainUserCheck,
 ) {
+  // TODO Check: can we amend this call to work
 
   val allGroups: List<Group>
-    get() = groupRepository.findAllByOrderByGroupName()
+    get() = runBlocking { groupRepository.findAllByOrderByGroupName().toList() }
```

Comment:
Is runBlocking actually blocking at this point?

Reply:
NO ACTION - removing method